modules/javafx.graphics/src/main/java/com/sun/javafx/sg/prism/NGNode.java

```diff
@@ -330,12 +330,12 @@ public void setTransformedBounds(BaseBounds bounds, boolean byTransformChangeOnl
         // cache needs to be regenerated. So we will not invalidate it here.
         if (dirtyBounds.isEmpty()) {
             dirtyBounds = dirtyBounds.deriveWithNewBounds(transformedBounds);
-            dirtyBounds = dirtyBounds.deriveWithUnion(bounds);
         } else {
             // TODO I think this is vestigial from Scenario and will never
             // actually occur in real life... (RT-23956)
             dirtyBounds = dirtyBounds.deriveWithUnion(transformedBounds);
         }
+        dirtyBounds = dirtyBounds.deriveWithUnion(bounds);
```

Comment:
Is it possible to create a unit test that fails with unmodified code but passes after the fix?

Reply:
Given the nature of the failure, an automated test would be difficult at best. Given the simplicity of the fix, testing it manually might be the best approach.